pkg/storage/ingest/reader.go

```diff
@@ -750,6 +761,16 @@ func newReaderMetrics(partitionID int32, reg prometheus.Registerer) readerMetric
 			Name: "cortex_ingest_storage_reader_fetches_total",
 			Help: "Total number of Kafka fetches received by the consumer.",
 		}),
+		fetchWaitDuration: promauto.With(reg).NewHistogram(prometheus.HistogramOpts{
+			Name:                        "cortex_ingest_storage_reader_fetch_wait_duration_seconds",
+			Help:                        "How long fetching a batch of records from the kafka client took to complete.",
+			NativeHistogramBucketFactor: 1.1,
+		}),
+		consumeLatency: promauto.With(reg).NewHistogram(prometheus.HistogramOpts{
+			Name:                        "cortex_ingest_storage_reader_consume_duration_seconds",
+			Help:                        "How long a request spent consuming a record batch from Kafka.",
```

Comment:
Have you considered rename "request" to "batch" (or something similar) and maybe renaming the metric too? What we track here is the consumption of a batch of fetches.

_Same comment applies to fetchWaitDuration`

Reply:
I opened https://github.com/grafana/mimir/pull/8217